backend/apps/beaver_api/beaver_api/beaver_api/test_views.py

```diff
@@ -0,0 +1,237 @@
+            code="print('Test Code 1')",
+            link_to_project="https://leetcode.com/problems/maximum-subarray/",
+            language=Language.objects.get(name="Python"),
+            code_content_hash=hashlib.sha256(
+                b"print('Test Code 1')"
+            ).hexdigest(),
+        )
+        code_document_1.tags.set(
+            [Tag.objects.get(name="tag1"), Tag.objects.get(name="tag2")]
+        )
+
+        code_document_2 = CodeDocument.objects.create(
+            id="19d5a9e6-f960-4014-a496-efa752d0855c",
+            title="Test document 2",
+            code="print('Test Code 2')",
+            link_to_project="https://leetcode.com/problems/unique-paths-ii/",
+            language=Language.objects.get(name="Python"),
+            code_content_hash=hashlib.sha256(
+                b"print('Test Code 2')"
+            ).hexdigest(),
+        )
+        code_document_2.tags.set(
+            [Tag.objects.get(name="tag1"), Tag.objects.get(name="tag2")]
+        )
+
+        self.url = reverse("code-document-bulk-delete")
+        self.valid_data = {
+            "ids": [
+                "f363f562-bc30-42aa-b96c-ff5c01c57e2c",
+                "19d5a9e6-f960-4014-a496-efa752d0855c",
+            ]
+        }
+
+    def test_bulk_delete_code_documents_with_valid_data(self):
+        self.assertEqual(CodeDocument.objects.count(), 2)
+
+        response = self.client.delete(
+            self.url, data=self.valid_data, format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(CodeDocument.objects.count(), 0)
+
+
+class CodeDocumentListViewTestCase(APITestCase):
+    def test_get_code_documents(self):
+        code_document_1 = CodeDocument.objects.create(
+            id="f363f562-bc30-42aa-b96c-ff5c01c57e2c",
+            title="Test document 1",
+            code="print('Test Code 1')",
+            link_to_project="https://leetcode.com/problems/search-a-2d-matrix/",
+            language=Language.objects.get(name="Python"),
+            code_content_hash=hashlib.sha256(
+                b"print('Test Code 1')"
+            ).hexdigest(),
+        )
+        code_document_1.tags.set(
+            [Tag.objects.get(name="tag1"), Tag.objects.get(name="tag2")]
+        )
+
+        code_document_2 = CodeDocument.objects.create(
+            id="19d5a9e6-f960-4014-a496-efa752d0855c",
+            title="Test document 2",
+            code="print('Test Code 2')",
+            link_to_project="https://leetcode.com/problems/partition-list/",
+            language=Language.objects.get(name="Python"),
+            code_content_hash=hashlib.sha256(
+                b"print('Test Code 2')"
+            ).hexdigest(),
+        )
+        code_document_2.tags.set(
+            [Tag.objects.get(name="tag1"), Tag.objects.get(name="tag2")]
+        )
+        self.url = reverse("code-documents")
+
+        response = self.client.get(self.url)
```

Comment:
Result of this call is always one element, instead of two. I don't see any abnormalities in the view.

Reply:
I see several issues here:
1. Please move all views and all code to the corresponding app 'code_api'. We are writing code to the main app only when we want to share the code between apps. Do not forget to update the URLs the code_api URLs already included to the main URL file, so you just need to move endpoints to the code_api urls.py (keep in mind theyalready have prefix)
2. In URLs you are routing two views to the same endpoints (my code_api rundom view and your List view), That is the reason you do not see it in the /docs endpoint and in the drf debug pages
3. You are using pagination, which means you are always getting one json (see the pic. below) you can find your data in the "results" section of the dict. 
<img width="277" alt="image" src="https://github.com/nikitazigman/beaver/assets/60356768/4bbb8dd6-2192-48f2-8d91-0d10a755e2f1">
You can read more about pagination here https://www.django-rest-framework.org/api-guide/pagination/

Reply:
Solved in recent commits.